modules/tinymce/src/core/main/ts/content/InsertContent.ts

```diff
@@ -41,9 +42,13 @@ const processValue = (value: string | DetailsWithContent): ProcessedValue => {
 };
 
 const insertAtCaret = (editor: Editor, value: string | DetailsWithContent): void => {
-  const result = processValue(value);
+  const { content, details } = processValue(value);
 
-  Rtc.insertContent(editor, result.content, result.details);
+  preProcessSetContent(editor, { content, format: 'html', set: false, selection: true, paste: details.paste }).each((args) => {
+    Rtc.insertContent(editor, content, details);
```

Comment:
This line is also incorrect, as we should be passing the pre-processed HTML, not the input HTML.
```suggestion
    Rtc.insertContent(editor, args.content, details);
```